src/iuse.cpp

```diff
@@ -8068,6 +8068,36 @@ int iuse::hairkit(player *p, item *it, bool, const tripoint&)
     return it->type->charges_to_use();
 }
 
+int iuse::washclothes( player *p, item *it, bool, const tripoint& )
+{
+    if( it->charges < it->type->charges_to_use() ) {
+        p->add_msg_if_player( _( "You need soap to use this." ) );
+    } else if ( g->m.has_nearby_water( g->u.pos() ) ) {
+        int pos = g->inv_for_filter( _( "Wash what?" ), []( const item & itm ) {
+            return itm.has_flag( "FILTHY" ); } );
+    item *mod = &( p->i_at( pos ) );
+
+    if( mod == NULL || mod->is_null() ) {
+        p->add_msg_if_player( m_info, _( "You don't have that item!" ) );
+        return 0;
+    }
+
+    if ( !mod->has_flag( "FILTHY" ) ) {
+        p->add_msg_if_player( m_info, _( "You can't do this!" ) );
+        return 0;
+    }
+
+    p->add_msg_if_player( _( "You washed your clothing." ) );
+    p->moves -= 3000;
```

Comment:
`p->mod_moves( -3000 )` would be better. Maybe even a player activity would make sense here (you can't wash your items at once and should be aware of perils around).